File: java/bypstest-api/src/byps/test/api/comp/IncompatibleChangeIF.java
package byps.test.api.comp;

import byps.BRemote;
import byps.RemoteException;

/**
 * This interface with incompatible changes.
 * @since 793
 */
public interface IncompatibleChangeIF extends BRemote {

  /**
   * Function with changed return type.
   * The return type was int in version 793. In 794 it has been changed to String.
   * Calling this function must cause an exception.
   * @return value
   * @throws RemoteException
   */
  public int changedReturnType() throws RemoteException;

  /**
   * Function with changed parameter type.
   * The parameter type was int in version 793. In 794 it has been changed to String.
   * Calling this function must cause an exception.
   * @param intParamChangedToString value
   * @throws RemoteException
   */
  public void changedParameterType(int intParamChangedToString) throws RemoteException;
  
  /**
   * Function with changed parameter class.
   * The class has been incompatibly changed in version 794.
   * Calling this function must cause an exception.
   * @param param value
   * @throws RemoteException
   */
  public void changedClass(IncompatibleChangeInfo param) throws RemoteException;
}

File: csharp/bypstest-ser/src-ser/byps/test/api/comp/IncompatibleChangeIF.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using byps;

namespace byps.test.api.comp
{
	
	/// <summary>
	/// This interface with incompatible changes.
	/// </summary>
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface IncompatibleChangeIF : BRemote {
	
		/// <summary>
		/// Function with changed return type.
		/// </summary>
		/// <remarks>
		/// The return type was int in version 793. In 794 it has been changed to String.
		/// Calling this function must cause an exception.
		/// </remarks>
		int ChangedReturnType();
		void ChangedReturnType(BAsyncResult<int> asyncResult) ;
		/// <summary>
		/// Function with changed return type.
		/// </summary>
		/// <remarks>
		/// The return type was int in version 793. In 794 it has been changed to String.
		/// Calling this function must cause an exception.
		/// </remarks>
		Task<int> ChangedReturnTypeAsync();
		
		/// <summary>
		/// Function with changed parameter type.
		/// </summary>
		/// <remarks>
		/// The parameter type was int in version 793. In 794 it has been changed to String.
		/// Calling this function must cause an exception.
		/// </remarks>
		void ChangedParameterType(int intParamChangedToString);
		void ChangedParameterType(int intParamChangedToString, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Function with changed parameter type.
		/// </summary>
		/// <remarks>
		/// The parameter type was int in version 793. In 794 it has been changed to String.
		/// Calling this function must cause an exception.
		/// </remarks>
		Task ChangedParameterTypeAsync(int intParamChangedToString);
		
		/// <summary>
		/// Function with changed parameter class.
		/// </summary>
		/// <remarks>
		/// The class has been incompatibly changed in version 794.
		/// Calling this function must cause an exception.
		/// </remarks>
		void ChangedClass(IncompatibleChangeInfo param);
		void ChangedClass(IncompatibleChangeInfo param, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Function with changed parameter class.
		/// </summary>
		/// <remarks>
		/// The class has been incompatibly changed in version 794.
		/// Calling this function must cause an exception.
		/// </remarks>
		Task ChangedClassAsync(IncompatibleChangeInfo param);
		
		
	}
}  // end namespace
